### PR TITLE
[DUOS-2023][risk=no] Adds download links for COL and IRB provided docs.

### PR DESCRIPTION
**Pull request by otchet-broad**
[DUOS-2023](https://broadworkbench.atlassian.net/browse/DUOS-2023) requests links are added to DARs where the applicant provided a COL or IRB document.
DarCollectionReview was modified to pass in the necessary information to ApplicationInformation to build links to the documents.
ajax was updated to include a download function for enabling document download with the required headers.
applicaiton_information.spec was updated to include tests to ensure the links were rendered correctly.

BEFORE
Screen Shot 2022-11-07 at 1 12 29 PM

AFTER
Screen Shot 2022-11-07 at 1 12 40 PM

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/dar_collection_review/DarCollectionReview.js
Comment (otchet-broad) [inline src/pages/dar_collection_review/DarCollectionReview.js:86]: In speaking with @JVThomas, any one of the collection of DARs should point to the proper referenceId that is needed to obtain the documents from the v2 endpoint.

File: src/libs/ajax.js
Comment (rushtong) [inline src/libs/ajax.js:263]: The method right above this one (line 252 `getDARDocument`), is what's called when we download the file from the DAR form. Currently, it uses the UUID of the file in GCS as the file name which is pretty poor UX. Can we either refactor that (see usages in `UploadLabelButton`) or file a ticket to do that?

File: src/pages/dar_collection_review/DarCollectionReview.js
Comment (rushtong) [reply src/pages/dar_collection_review/DarCollectionReview.js:86]: That is correct. Alternatively, we can add parallel APIs at the collection level that do the same thing. I'm fine with this approach.

**Review by rushtong — APPROVED** [review]
This looks and works great, thank you! One suggestion inline that doesn't have to hold up the PR and can be handled separately.

File: src/libs/ajax.js
Comment (otchet-broad) [reply src/libs/ajax.js:263]: I'm glad you caught this.  I want to do a bigger refactor here and recommend we make a new ticket for that.  May I do that?

Wherever we're using "createObjectURL" and we're not using revokeObjectURL to remove it, we're potentially leaking memory.  The `fileDownload` utility I've used does that for us automatically.  See: https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL


This happens in `UploadLabelButton`, `downloadDataSets`, and in utils `download`

And to your point, I'd also like to make sure we're providing file names whenever possible.

Comment (otchet-broad) [reply src/libs/ajax.js:263]: See [DUOS-2190](https://broadworkbench.atlassian.net/browse/DUOS-2190)

Comment (rushtong) [reply src/libs/ajax.js:263]: > I'm glad you caught this. I want to do a bigger refactor here and recommend we make a new ticket for that. May I do that?

Absolutely. See also https://broadworkbench.atlassian.net/browse/DUOS-2188 where I'd like to do a much better job of handling DAR files.

**Review by quazi-h — APPROVED** [review]
The changes look good and I can see IRB and Collaboration Letter download links on DARs that have them. When testing locally, I wasn't able to download anything from DAR-586 but I think that might be because it's a Read-Only for me? After creating a new DAR with IRB/Collaboration files, I was able to download them from the dar_collection_review page.

The console revealed that we were hitting 403 errors when attempting to use those links on a DAR that the current user did not create:
`GET https://local.broadinstitute.org:3000/api/dar/v2/e9340c8c-f4dc-48d4-8f54-627d8a5d1754/irbDocument 403 (Forbidden)`
This is because of the checkAuthorizedUpdateUser in DataAccessRequestResourceVersion2.java. We may need to revise the backend logic to implement a new check that looks to see if the current user is in a list of approved users who should have access to that particular DAR.

**Comment by otchet-broad** [comment]
Great find @quazi-broad!  I'm going to create a PR to address the backend issue.

After speaking with @rushtong, we determined that members, chairs and admins should be able to see the contents.

I'll park this PR until the backend changes are approved.

**Pull request merged**
